.github/workflows/inspektor-gadget.yml

```diff
@@ -427,7 +427,7 @@ jobs:
       with:
         context: /home/runner/work/inspektor-gadget/inspektor-gadget/
         file: /home/runner/work/inspektor-gadget/inspektor-gadget/Dockerfiles/ig.Dockerfile
-        outputs: type=registry,name=${{ steps.set-repo-determine-image-tag.outputs.container-repo }},push=true,push-by-digest=true
+        outputs: type=registry,name=${{ steps.set-repo-determine-image-tag.outputs.container-repo }},push=true,push-by-digest=true,annotation-index.org.opencontainers.image.description=Inspektor Gadget is a collection of tools (or gadgets) to debug and inspect Kubernetes resources and applications. This image only includes the ig binary, a standalone tool to run the gadgets.
```

Comment:
Meanwhile, I rephrased it without the comma.

Tested in my fork:
https://github.com/alban/inspektor-gadget/actions/runs/5597096480/jobs/10234999557

But the description was not added correctly on the package page:
https://github.com/alban/inspektor-gadget/pkgs/container/ig/110774229?tag=citest-alban_image_labels

The labels are added correctly:
```
$ docker pull ghcr.io/alban/ig:citest-alban_image_labels@sha256:2ffe57dd98618377dd73a03ccfe1e27407ce43aff7b67f65eba2be424e0ceab5
$ docker inspect ghcr.io/alban/ig:citest-alban_image_labels@sha256:2ffe57dd98618377dd73a03ccfe1e27407ce43aff7b67f65eba2be424e0ceab5 | jq '.[0].Config.Labels'
{
  "org.opencontainers.image.description": "Inspektor Gadget is a collection of tools (or gadgets) to debug and inspect Kubernetes resources and applications. This image only includes the ig binary, a standalone tool to run the gadgets.",
  "org.opencontainers.image.documentation": "https://inspektor-gadget.io/docs",
  "org.opencontainers.image.licenses": "Apache-2.0",
  "org.opencontainers.image.source": "https://github.com/inspektor-gadget/inspektor-gadget",
  "org.opencontainers.image.title": "Inspektor Gadget ig tool"
}
```

But not the annotation:
```
$ skopeo inspect --raw docker://ghcr.io/alban/ig:citest-alban_image_labels | less
```

I didn't find in the docker/actions-toolkit where the annotation is supposed to be read from the `outputs:` field.

The annotation output is implemented in moby/buildkit v0.11.0-rc1 by https://github.com/moby/buildkit/commit/a6a114a1a476c99c2501aa34811159b849df4005. The GitHub Action uses:
```
/usr/bin/docker buildx version
github.com/docker/buildx 0.11.1+azure-1 b4df08551fb12eb2ed17d3d9c3977ca0a903415a
```

I checked this commit:
```
$ git clone https://github.com/docker/buildx.git
$ cd buildx
$ git checkout b4df08551fb12eb2ed17d3d9c3977ca0a903415a
$ grep moby/buildkit go.mod
	github.com/moby/buildkit v0.11.0-rc3.0.20230609092854-67a08623b95a
$ cd ..
$ git clone https://github.com/moby/buildkit.git
$ cd buildkit
$ git checkout 67a08623b95a
$ git grep annotation-index -- docs/
docs/annotations.md:59:        --output "type=image,name=target,annotation-index.org.opencontainers.image.title=Target Image"
```

So the version used seems to have the support for annotation-index, but it still didn't work.

Reply:
It seems multi arch docker builds with a matrix need to be done in 2 steps:
1. An action in a matrix (arm64, amd64) to build for the specific arch
2. Another action outside of the matrix to build the image manifest manually with `docker buildx imagetools create` and and pushing the result

The description annotation should be done in step 2. The step 1 is not preparing the global manifest so it cannot contain the annotation.

https://docs.docker.com/build/ci/github-actions/multi-platform/#distribute-build-across-multiple-runners

See also
* https://github.com/docker/build-push-action/issues/846#issuecomment-1543634155
* https://github.com/docker/docs/pull/17180/files

Reply:
> https://github.com/inspektor-gadget/inspektor-gadget/blob/main/.github/workflows/inspektor-gadget.yml#L507

Interesting that they documented it now, back in time when we wanted to use this approach I couldn't find any documentation but we use the same [approach](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/.github/workflows/inspektor-gadget.yml#L507). 

> The description annotation should be done in step 2

Is is possible to set annotations using `docker buildx imagetools create` ?

Reply:
I attempted this but it panics:
https://github.com/alban/inspektor-gadget/actions/runs/5600745874/jobs/10244060496
```
$ docker buildx imagetools create -t ghcr.io/alban/ig:citest-alban_image_labels -f descr.json
#1 [internal] pushing ghcr.io/alban/ig:citest-alban_image_labels
#1 0.001 pushing  to ghcr.io/alban/ig:citest-alban_image_labels
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1162736]

goroutine 6 [running]:
io.(*PipeWriter).Write(...)
	/usr/local/go/src/io/pipe.go:165
github.com/containerd/containerd/remotes/docker.(*pushWriter).Commit(0xc0000c11f0, {0x0?, 0x26bbc00?}, 0x0, {0x0, 0x0}, {0xc1261d43cfeb08c3?, 0xd1db55d?, 0x38319a0?})
	/build/moby-buildx/.gopath/src/github.com/docker/buildx/vendor/github.com/containerd/containerd/remotes/docker/pusher.go:441 +0x76
github.com/containerd/containerd/content.Copy({0x26ded48, 0xc0005f2de0}, {0x26e3328, 0xc0000c11f0}, {0x26bb7c0, 0xc000794300}, 0x0, {0x0, 0x0}, {0x0, ...})
	/build/moby-buildx/.gopath/src/github.com/docker/buildx/vendor/github.com/containerd/containerd/content/helpers.go:186 +0x3df
github.com/docker/buildx/util/imagetools.(*Resolver).Push(0xc000133f70?, {0x26deca0?, 0xc0001143c0?}, {0x26ccd48, 0xc0005e51c0}, {{0xc0003f18f0, 0x27}, {0x0, 0x0}, 0x0, ...}, ...)
	/build/moby-buildx/.gopath/src/github.com/docker/buildx/util/imagetools/create.go:175 +0x29c
github.com/docker/buildx/commands/imagetools.runCreate.func2.1({0x26ce5a0?, 0xc000010828})
	/build/moby-buildx/.gopath/src/github.com/docker/buildx/commands/imagetools/create.go:200 +0x4d0
github.com/docker/buildx/util/progress.Wrap({0xc0005e2e80, 0x32}, 0xc000469d70, 0xc000530e90)
	/build/moby-buildx/.gopath/src/github.com/docker/buildx/util/progress/progress.go:47 +0x2dc
github.com/docker/buildx/commands/imagetools.runCreate.func2()
	/build/moby-buildx/.gopath/src/github.com/docker/buildx/commands/imagetools/create.go:[183](https://github.com/alban/inspektor-gadget/actions/runs/5600745874/jobs/10244060496#step:5:184) +0x29a
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/build/moby-buildx/.gopath/src/github.com/docker/buildx/vendor/golang.org/x/sync/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/build/moby-buildx/.gopath/src/github.com/docker/buildx/vendor/golang.org/x/sync/errgroup/errgroup.go:72 +0xa5
```

The panic is reproducible.

Reply:
It should be connected, this is from the GitHub Action, see link above.

Reply:
What do you think about using [crane](https://github.com/imjasonh/setup-crane) for this? I tried running it locally and it seem to have worked [fine](https://github.com/mqasimsarfraz/inspektor-gadget/pkgs/container/inspektor-gadget/111105723?tag=latest):

```
crane mutate  ghcr.io/mqasimsarfraz/inspektor-gadget:latest   --annotation=org.opencontainers.image.description="Inspektor Gadget is a collection of tools (or gadgets) to debug and inspect Kubernetes resources and applications. This image only includes the ig binary, a standalone tool to run the gadgets"
```

Reply:
> > it seem to have worked [fine](https://github.com/mqasimsarfraz/inspektor-gadget/pkgs/container/inspektor-gadget/111105723?tag=latest)
> 
> Is this a multiarch image?

Sadly it seems the image isn't multiarch anymore :(

Reply:
> I think the overall here is pretty damn too much complicated for no real plus-value.

100%, I would also vote to avoid it. 

> Error: mutating annotations on an index is not yet supported

Perhaps I was using an older version of `crane` and it was missing this [fix](https://github.com/google/go-containerregistry/pull/1056/commits/f99b6a83855b137df9511e8dba04038230d93339). I upgraded to latest crane and now seeing same error as you.

Reply:
> I think the overall here is pretty damn too much complicated for no real plus-value.
> I would advise to first focus on the "in Dockerfile" LABEL, which seems quite simple to do.
> Then, we need to get a better understanding of what we are missing there to then have this reflected on the package page.

Ok, I will update this PR to do only the "in Dockerfile" LABEL. The rest will be for a future PR. But this means the GitHub package page will not contain any description.